modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1131,10 +1149,80 @@ CV__DNN_INLINE_NS_BEGIN
      *           It differs from the above function only in what argument(s) it accepts.
      */
     CV_EXPORTS void blobFromImages(InputArrayOfArrays images, OutputArray blob,
-                                   double scalefactor=1.0, Size size = Size(),
+                                   const Scalar& scalefactor = 1.0, Size size = Size(),
                                    const Scalar& mean = Scalar(), bool swapRB=false, bool crop=false,
                                    int ddepth=CV_32F);
 
+    /**
+     * @brief Enum of image preprocessing mode.
+     * @see ImagePParam
+     */
+    enum PreprocessMode
+    {
+        DNN_PP_CENTER = 0,    // !< Default
+        DNN_PP_LETTERBOX = 1, // !< Resize image to the desired size while preserving the aspect ratio of original image.
+    };
+
+    /** @brief Image preprocessing parameters.
+     *
+     * It includes all possible image preprocessing operations and corresponding parameters.
+     *
+     * @see blobFromImageParam
+     * @note
+     * The order and usage of `scalefactor`, `size`, `mean`, `swapRB`, `crop` and `ddepth` are consistent
+     * with the function of @ref blobFromImage.
+    */
+    struct CV_EXPORTS_W_SIMPLE ImagePParam
+    {
+        CV_WRAP ImagePParam();
+        CV_WRAP ImagePParam(const Scalar& scalefactor, const Size& size = Size(), const Scalar& mean = Scalar(),
+                            bool swapRB = false, bool crop = false, int ddepth = CV_32F,
+                            DataLayout datalayout = DNN_LAYOUT_NCHW, PreprocessMode mode = DNN_PP_CENTER);
+
+        CV_PROP_RW Scalar scalefactor; //!< scalefactor multiplier for input image values.
```

Comment:
Does `(1, 0, 0, 0)` interpreted internally as `(1,1,1,1)`?

Reply:
That's a workaround for me.